addons/stock/models/stock_quant.py

```diff
@@ -312,6 +312,15 @@ def _load_records_write(self, values):
         """ Only allowed fields should be modified """
         return super(StockQuant, self.with_context(inventory_mode=True))._load_records_write(values)
 
+    @api.model
+    def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True):
+        """
+        Override to display the inventory_quantity_auto_apply:sum in group by of the list view
+        """
+        if 'inventory_quantity_auto_apply' in fields and 'quantity' not in fields:
+            fields.append('inventory_quantity_auto_apply:sum')
```

Comment:
I would do this only if `product_id` is in the `groupby` parameter to avoid non sens quantity like this 
![image](https://github.com/user-attachments/assets/68df6eaf-b738-4ab4-aed7-d6a6d7e0c116)